src/overlays/ovl_i2/fox_sx.c

```diff
@@ -1,81 +1,1986 @@
+    Audio_PlaySfx(0x09000002U, &actor->sfxPos, 0U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+}
+
+void func_i2_80193A30(Player* player) {
+    s32 i;
+    Vec3f src;
+    Vec3f dest;
+    f32 x;
+    f32 y;
+    f32 z;
+    f32 xzDeg;
+    f32 xyzDeg;
+
+    switch (player->unk_1D0) {
+        case 0:
+            D_80177A80 = 0;
+            for (i = 0; i < 5; i++) {
+                func_i2_80193800(&gActors[5 + i], i);
+            }
+            player->unk_1D0 = 1;
+            player->camEye.x = D_80177978 = 100.0f;
+            player->camEye.y = D_80177980 = 0.0f;
+            player->camEye.z = D_80177988 = -200.0f;
+            player->camAt.x = D_801779A0 = 0.0f;
+            player->camAt.y = D_801779B8 = 0.0f;
+            player->camAt.z = D_801779C0 = -1000.0f;
+            player->unk_034 = -5.0f;
+            player->unk_0D0 = 0.0f;
+            D_80178340 = 0xFF;
+            /* fallthrough */
+        case 1:
+            if (D_80177A80 < 3) {
+                D_80178340 = 0xFF;
+            }
+            D_80177A48[0] = 0.5f;
+            D_801779A0 = gActors[5].obj.pos.x;
+            D_801779C0 = gActors[5].obj.pos.z;
+            player->unk_034 -= 0.1f;
+            if (D_80177A80 == 0x8C) {
+                x = gActors[5].obj.pos.x;
+                y = gActors[5].obj.pos.y - (player->camEye.y + 50.0f);
+                z = gActors[5].obj.pos.z - (player->camEye.z + 20.0f);
+                xzDeg = Math_RadToDeg(Math_Atan2F(x, z));
+                xyzDeg = Math_RadToDeg(-Math_Atan2F(y, sqrtf(SQ(x) + SQ(z))));
+                Matrix_RotateY(gCalcMatrix, M_DTOR * xzDeg, 0);
+                Matrix_RotateX(gCalcMatrix, M_DTOR * xyzDeg, 1);
+                src.y = src.x = 0.0f;
+                src.z = 100.0f;
+                Matrix_MultVec3fNoTranslate(gCalcMatrix, &src, &dest);
+                func_8006EEFC(0x64, 0.0f, player->camEye.y + 50.0f, player->camEye.z + 20.0f, dest.x, dest.y, dest.z,
+                              xyzDeg, xzDeg, 0.0f);
+            }
+            if (D_80177A80 == 0x8F) {
+                Object_Kill(&gPlayerShots->obj, &gPlayerShots->sfxPos);
+                func_8007BFFC(gActors[5].obj.pos.x, gActors[5].obj.pos.y, gActors[5].obj.pos.z, 0.0f, 0.0f, 0.0f, 3.0f,
+                              0x28);
+                func_8007D2C8(gActors[5].obj.pos.x, gActors[5].obj.pos.y, gActors[5].obj.pos.z, 6.0f);
+                func_8007B344(gActors[5].obj.pos.x, gActors[5].obj.pos.y, gActors[5].obj.pos.z, 5.0f, 5);
+                Object_Kill(&gActors[5].obj, &gActors[5].sfxPos);
+                for (i = 0; i < 20; i++) {
+                    func_80079618(gActors[5].obj.pos.x, gActors[5].obj.pos.y, gActors[5].obj.pos.z, 0.5f);
+                }
+                func_8007A6F0(&gActors[5].obj.pos, 0x2903B009);
+                player->unk_1D0 += 1;
+                D_80177A48[0] = 0.1f;
+            }
+            break;
+
+        case 2:
+            if (D_80177A80 == 0xAA) {
```

Comment:
```suggestion
            if (D_80177A80 == 170) {
```
`D_80177A80` is also a frame counter, so should be decimal.